### PR TITLE
buffered logging for service, revendor

### PR DESCRIPTION
**Pull request by joshblum**
vendors changes from: https://github.com/keybase/go-logging/pull/9

### CODE REVIEW

File: go/keybase/main.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ func keybaseExit(exitCode int) {
 }
 
 func main() {
+	logger.EnableBufferedLogging()
```
Comment (mmaxim) [inline go/keybase/main.go:50]: Don't need this

File: go/logger/buffer.go [hunk 2]
```diff
@@ -88,13 +88,10 @@ func (writer *autoFlushingBufferedWriter) backgroundFlush() {
 
 			writer.backupWriter.Flush()
 		case <-writer.shutdown:
-			writer.timer.shutdownCh <- struct{}{}
+			close(writer.timer.shutdownCh)
```
Comment (mmaxim) [inline go/logger/buffer.go:91]: Why are these changes here?

File: go/keybase/main.go (hunk 1 above)
Comment (joshblum) [reply go/keybase/main.go:50]: why not? we only call it in kbfs dont' we want buffered logging in the service?

File: go/logger/buffer.go (hunk 2 above)
Comment (joshblum) [reply go/logger/buffer.go:91]: required to make shutdown re-entrant

Comment (mmaxim) [reply go/logger/buffer.go:91]: I think you likely need this because of the other change which I think you should remove

File: go/keybase/main.go (hunk 1 above)
Comment (mmaxim) [reply go/keybase/main.go:50]: It gets turned on in `SetLogFileConfig`

File: go/logger/buffer.go (hunk 2 above)
Comment (joshblum) [reply go/logger/buffer.go:91]: ok cool

**Comment by joshblum** [comment]
ci is dead, this is unrelated an was previously green.
